hack/generator/pkg/astmodel/storage_conversion_factories.go

```diff
@@ -849,53 +843,3 @@ func createTypeDeclaration(name TypeName, generationContext *CodeGenerationConte
 	packageName := generationContext.MustGetImportedPackageName(name.PackageReference)
 	return astbuilder.Selector(dst.NewIdent(packageName), name.Name())
 }
-
-func zeroValue(t Type, types Types) string {
-	if isTypeOptional(t) {
-		return "nil"
-	}
-
-	if name, isName := AsTypeName(t); isName {
-		// We've got a type name, need to see what it resolves to
-		actualType, err := types.FullyResolve(name)
-		if err != nil {
-			// This should never happen
-			panic(err)
-		}
-
-		enumType, isEnum := AsEnumType(actualType)
-		if isEnum {
-			// Zero value for an enum is the zero value of the base type
-			return zeroValue(enumType.baseType, types)
-		}
-
-		primitiveType, isPrimitive := AsPrimitiveType(actualType)
-		if isPrimitive {
-			return zeroValue(primitiveType, types)
-		}
-
-		// Otherwise default to an empty instantiation
-		return fmt.Sprintf("%s{}", name.Name())
```

Comment:
This is doing the right thing for ObjectType, I think. (Although does it need a package qualification?)

Reply:
Updated `TypeName.AsZero()` to use a package name if needed.